decidim-api/lib/decidim/api/test/component_context.rb

```diff
@@ -42,3 +42,149 @@
     )
   end
 end
+
+shared_examples "with resource visibility" do
+  let(:process_space_factory) { :participatory_process }
+
+  context "when space is published" do
+    let!(:participatory_process) { create(process_space_factory, :published, :with_steps, organization: current_organization) }
+
+    context "when component is published" do
+      let!(:current_component) { create(component_factory, :published, participatory_space: participatory_process) }
+
+      context "when the user is admin" do
+        let!(:current_user) { create(:user, :admin, :confirmed, organization: current_organization) }
+
+        it { expect(response["participatoryProcess"]["components"].first[lookout_key]).to eq(query_result) }
+      end
+
+      context "when user is anonymous" do
+        let!(:current_user) { nil }
+
+        it { expect(response["participatoryProcess"]["components"].first[lookout_key]).to eq(query_result) }
+      end
+
+      context "when user is normal user" do
+        let!(:current_user) { create(:user, :confirmed, organization: current_organization) }
+
+        it { expect(response["participatoryProcess"]["components"].first[lookout_key]).to eq(query_result) }
+      end
+    end
+
+    context "when component is not published" do
+      let!(:current_component) { create(component_factory, :unpublished, participatory_space: participatory_process) }
+
+      context "when the user is admin" do
+        let!(:current_user) { create(:user, :admin, :confirmed, organization: current_organization) }
+
+        it { expect(response["participatoryProcess"]["components"].first[lookout_key]).to eq(query_result) }
+      end
+
+      context "when user is anonymous" do
+        let!(:current_user) { nil }
+
+        it { expect(response["participatoryProcess"]["components"].first).to be_nil }
```

Comment:
```suggestion
        it "should not be visible" do
          expect(response["participatoryProcess"]["components"].first).to be_nil
        end
```

Reply:
Since there are multiple places where this needs to be changed, i've done it in commit https://github.com/decidim/decidim/commit/89fee855e1bc6cdb19b59b464fcdf5b143bd0a53